docs/api/tiler.rst

```diff
@@ -1,5 +1,48 @@
+#. ``prefix``: a prefix to be added at the beginning of the tiles' filename (optional, default is the empty string);
+#. ``suffix``: a suffix to be added to the end of the tiles' filename (optional, default is ``.png``).
+
+All ``histolab`` extractors define the ``extract`` method, used to start the extraction process on the ``Slide`` passed as parameter.
+Each extractor will automatically (i) find the largest tissue box in the WSI; (ii) generate the tiles within the tissue region; (iii) save all generated tiles or only the informative ones if the attribute ``check_tissue`` is set to True.
+
+RandomTiler
+-----------
+The `RandomTiler <tiler.html#src.histolab.tiler.RandomTiler>`_ extractor allows for the extraction of tiles picked at random within the largest tissue box. Since there is no intrinsic upper bound of the number of the tiles that could be extracted (no overlap check is performed), the number of wanted tiles must be specified.
+
+In addition to 1-6, the `RandomTiler <tiler.html#src.histolab.tiler.RandomTiler>`_ constructor requires as two additional parameters the number of tiles requested (``n_tiles``), and the random seed (``seed``), to ensure reproducibility between different runs on the same WSI. Note that less than ``n_tiles`` could be extracted from a slide with not enough tissue pixels and a lot of background, which is checked when the parameter ``check_tissue`` is set to True.
+``n_tiles`` will be interpreted as the upper bound of the number of tiles requested: it might not be possible to extract ``n_tiles`` tiles from a slide with a little tissue sample and a lot of background.
+
+The extraction procedure will (i) find the largest tissue area in the WSI; (ii) generate ``n_tiles`` random tiles; (iii) save only the tiles with enough tissue if the attribute ``check_tissue`` was set to True, save all the generated tiles otherwise.
+
+GridTiler
+----------
+A second basic approach consists in extracting all the tiles in the largest tissue region in the WSI. This strategy is implemented in the `GridTiler <tiler.html#src.histolab.tilerGridTiler>`_ class. The additional ``pixel_overlap`` parameter specifies the number of overlapping pixels between two adjacent tiles, i.e. tiles are cropped by using a sliding window with stride s defined as:
+
+.. math::
+
+     s=(w - \mathrm{pixel\_overlap}) \cdot (h - \mathrm{pixel\_overlap})
+
+where w and h are customizable parameters defining the width and the height of the resulting tiles.
+Calling the ``extract`` method on the `GridTiler <tiler.html#src.histolab.tiler.GridTiler>`_ instance will automatically (i) find the largest tissue area in the WSI; (ii) generate all the tiles according to the grid structure; (iii) save only the tiles with "enough tissue" if the attribute ``check_tissue`` was set to True, save all the generated tiles otherwise.
```

Comment:
```suggestion
Calling the ``extract`` method on the `GridTiler <tiler.html#src.histolab.tiler.GridTiler>`_ instance will automatically (i) find the regions to extract tiles from, defined by the binary mask object; (ii) generate all the tiles according to the grid structure; (iii) save only the tiles with "enough tissue" if the attribute ``check_tissue`` was set to True, save all the generated tiles otherwise.
```